src/sentry/interfaces/base.py

```diff
@@ -96,15 +105,43 @@ def __getattr__(self, name):
         return self._data[name]
 
     def __setattr__(self, name, value):
-        if name == '_data':
-            self.__dict__['_data'] = value
+        if name in ('_data', '_meta'):
+            self.__dict__[name] = value
         else:
             self._data[name] = value
 
     @classmethod
-    def to_python(cls, data):
+    def _to_python(cls, data, meta, **kwargs):
         return cls(**data)
 
+    @classmethod
+    def to_python(cls, data, meta=None, **kwargs):
+        """Creates a python interface object from the given raw data. By
+        default, this uses all keys passed in as `data`.
+
+        To override this behavior, implement `_to_python`.
+        """
+
+        if meta is None:
+            meta = Meta()
+
+        try:
+            if meta.get_errors():
+                instance = cls()
+            else:
+                instance = cls._to_python(data, meta, **kwargs)
+        except Exception as e:
+            # TODO(ja): Remove this after switching to Rust normalization
+            if not isinstance(e, InterfaceValidationError):
+                logger.error('Discarded invalid value for interface: %s (%r)',
```

Comment:
Will this happen when users send us invalid data as part of their events? Or when we have misaligned validation logic internally?

Reply:
`InterfaceValidationError` indicates invalid data as part of events. The next steps after this PR is to remove all of them and replace them with `meta.add_error`. After that, we will completely remove `_to_python` and rely on the Rust normalization only.

Reply:
Will these end up in sentry as issues? Customers send us lot of bogus data in the `release` and `environment` fields.

Reply:
All non-`InterfaceValidationError`s should end up in Sentry since they bugs in our `to_python` implementation, just like they did before:

https://github.com/getsentry/sentry/blob/a4ce1e54802c332634520e3b93ff5b093a6a0abe/src/sentry/event_manager.py#L556-L563

Reply:
On a side note, neither `release` nor `environment` will ever run through this code since they are no "interfaces". JSON schema validation has already run at that point (enforcing strings and pattern) and need to be handled differently.

I guess it's important to point out that all code surrounding `EventManager.normalize` and `Interface.to_python` will be gone within a month. This PR is intended to bring the rest of the Sentry code base up to speed.